### ISSUE TITLE
Add ability to use placeholders in the query matcher

### ISSUE THREAD

**Issue by UnwrittenFun** [caddyserver/caddy#3520]
As far as I can tell, you cannot use placeholders in the value (or key) with the query matcher.

I would like to be able to do, e.g:
```json
"match": [
  {
    "query": {
      "myVar": ["{http.vars.myVar}"]
    }
  }
],
```

But currently is not possible with the built-in query matcher. I will be getting around this by implementing a plugin with this supported for now, but official support would be 👍 

Personally I don't need support in the keys of the query, but maybe a useful addition as well?

**Comment by francislavoie** [comment]
Should be pretty easy to write a PR + tests for it, you're welcome to do so!

Relevant code is here:

https://github.com/caddyserver/caddy/blob/master/modules/caddyhttp/matchers.go#L376

Look at the other matchers and how they grab a replacer instance which can be invoked before matching.

**Comment by mholt** [comment]
Thanks for the PR!

**Comment by francislavoie** [comment]
FYI this ended up getting released in Caddy v2.1.1 because it could almost be considered a bugfix instead of a new feature and it was more convenient to include it than not to since we needed to make a new release to fix a separate issue.
